stl/inc/flat_set

```diff
@@ -513,8 +513,8 @@ private:
     template <class _Other>
         requires _Keylt_transparent || is_same_v<_Other, _Kty>
     _NODISCARD const_iterator _Find(const _Other& _Val) const {
-        const iterator _End   = end();
-        const iterator _Where = lower_bound(_Val);
+        const const_iterator _End   = end();
+        const const_iterator _Where = lower_bound(_Val);

```

Comment:
Rationale: `iterator` is not convertible to `const_iterator` (return type) in many cases.

Reply:
Can you explain a bit? I thought it was required to have `iterator` convertible to `const_iterator` https://eel.is/c++draft/container.reqmts#6

Reply:
This will fail; is this a bug in the current implementation?
```
    vector<int> i;
    auto nonc = i.begin();
    nonc = i.cbegin();
```

Reply:
Ugh, my mistake; `const_iterator` is not required to convert to `iterator`. (The fix is correct.)

Reply:
Ugh, my mistake; correction: const_iterator is not required to convert to iterator.
Update: the 3rd commit doesn't "fix" anything (but can be seen as nitpick/enhancement); there are real iterator fixes, see 4th commit.